packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -348,6 +358,14 @@ export async function runNpmInstall(
       env.PATH = `/node16/bin-npm7:${env.PATH}`;
       console.log('Detected `package-lock.json` generated by npm 7...');
     }
+  } else if (cliType === 'pnpm') {
+    await execCommand('pnpm config set store-dir .pnpm-store');
```

Comment:
Does this config need to be executed each time? Perhaps this should happen once in the build image?

Reply:
as far as I can tell this is a per-project configuration. `pnpm config` is actually an alias to `npm config` if I understand their docs correctly. Thus, I believe you are correct and that this command only needs to be executed once.

> Perhaps this should happen once in the build image

So should this be executed via docker? Or is there another place in the code base for it?

> Or could we use an env var instead? 

We could create an env var for the path so users can override, but we should default it to `.pnpm-store`. @jaredpalmer mentioned an idea of putting it inside of `node_modules` potentially too

Reply:
> this is a per-project configuration

Is it per-project or per-machine? Where is the config stored?

Reply:
Ahh good point, per-machine

Reply:
> Where is the config stored?

it has the same behavior as `.npmrc`. If a global one exists, it uses that, otherwise it looks for one in the project scope/directory

Reply:
Is the resolution here a change in the build-container Docker file? cc @jaredpalmer

Reply:
pnpm is per-machine is by default IIRC, but I'm not personally familiar enough with our build container and infra to understand security implications of ~/.pnpm-store. Looping in @gudmundur @igorklopov for their opinions.

I think for now, to unblock and unlock ARR, we can do this per-project even though it's obviously suboptimal from what is possible

Reply:
From my understanding, the default store is in the home directory at `~/.pnpm-store` and the command you're running would put the store in the local project directory ` ./.pnpm-store`.

https://pnpm.io/npmrc#store-dir

This behavior is actually worse than default because you're creating a new store for every project and the store isn't cached so you're guaranteeing a new store every time.

What is the goal of `pnpm config set store-dir .pnpm-store`? Can it be removed?

Reply:
We can place it somewhere here depending on what it is going to contain https://github.com/vercel/api/blob/412e6f6814e85ad8045d25a9e1736e791d0e228e/scripts/build-container/src/utils/npm-utils.ts#L252. Either we run `pnpm config set store-dir .pnpm-store` here or add a corresponding config line or something similar.

Reply:
Per-machine equals per-deployment so i don't see any security issues, given `~/.pnpm-store` is empty at start.